src/domain/matomo/matomo.tsx

```diff
@@ -0,0 +1,43 @@
+import {
+  createInstance as createMatomoInstance,
+  MatomoProvider,
+  useMatomo,
+} from "@datapunt/matomo-tracker-react";
+import { useRouter } from "next/router";
+import React, { useEffect } from "react";
+
+const getMatomoUrlPath = (path: string) =>
+  `${process.env.NEXT_PUBLIC_MATOMO_URL_BASE}${path}`;
+
+const matomoInstance = createMatomoInstance({
+  disabled: process.env.NEXT_PUBLIC_MATOMO_ENABLED !== "true",
+  urlBase: process.env.NEXT_PUBLIC_MATOMO_URL_BASE as string,
+  srcUrl: getMatomoUrlPath("piwik.min.js"),
+  trackerUrl: getMatomoUrlPath("tracker.php"),
+  siteId: Number(process.env.NEXT_PUBLIC_MATOMO_SITE_ID),
```

Comment:
We've accessed environment variables through `Config` so that they are easier to refactor.